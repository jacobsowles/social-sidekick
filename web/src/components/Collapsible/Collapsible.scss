.Collapsible {
  background: #ffffff;
  border: 1px solid rgba(0, 0, 0, 0.1);

  h1 {
    font-size: 2em;
    line-height: unset;
  }

  .Collapsible__contentInner {
    margin: 20px;
  }

  .Collapsible__trigger {
    align-content: center;
    display: flex;
    padding: 20px;
    position: relative;

    &:hover {
      cursor: pointer;
    }

    &:after {
      content: '\f107';
      display: block;
      font-family: 'FontAwesome';
      font-size: 1.5em;
      position: absolute;
      right: 20px;
      top: 25px;
      transition: transform 300ms;
    }

    &.is-open {
      &:after {
        transform: rotateZ(180deg);
      }
    }

    &.is-disabled {
      opacity: 0.5;
      background-color: grey;
    }

    svg {
      margin-right: 10px;
    }
  }
}
