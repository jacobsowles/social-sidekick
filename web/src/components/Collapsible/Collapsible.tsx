import React, { FunctionComponent } from 'react';
import Collapsible from 'react-collapsible';

import './Collapsible.scss';

//// BOX ////

interface BoxProps {
  className?: string;
}

const Box: FunctionComponent<BoxProps> = ({ children, className, ...rest }) => {
  // TODO: Do some validation on this. Require the first child to be a trigger, etc.
  const trigger = (children as any[])[0];
  const body = (children as any[])[1];

  return (
    <Collapsible className={className} transitionTime={200} trigger={trigger} {...rest}>
      {body}
    </Collapsible>
  );
};

Box.displayName = 'Collapsible.Box';

//// TRIGGER ////

const Trigger: FunctionComponent = ({ children }) => {
  return <>{children}</>;
};

Trigger.displayName = 'Collapsible.Trigger';

//// BODY ////

const Body: FunctionComponent = ({ children }) => {
  return <>{children}</>;
};

Body.displayName = 'Collapsible.Body';

export default { Box, Trigger, Body };
