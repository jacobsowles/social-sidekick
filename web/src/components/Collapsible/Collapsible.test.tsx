import React from 'react';

import { init, shallow, toJson } from '@tests/component-test-base';
import Collapsible from '@components/Collapsible';

describe('Collapsible', () => {
  init(
    () => (
      <Collapsible.Box>
        <Collapsible.Trigger />
        <Collapsible.Body />
      </Collapsible.Box>
    ),
    'div'
  );

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });
});
