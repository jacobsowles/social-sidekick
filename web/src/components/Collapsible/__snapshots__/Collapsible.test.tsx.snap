// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible should render the component without crashing 1`] = `
<Collapsible
  className=""
  classParentString="Collapsible"
  contentInnerClassName=""
  contentOuterClassName=""
  easing="linear"
  lazyRender={false}
  onClose={[Function]}
  onClosing={[Function]}
  onOpen={[Function]}
  onOpening={[Function]}
  open={false}
  openedClassName=""
  overflowWhenOpen="hidden"
  tabIndex={null}
  transitionCloseTime={null}
  transitionTime={200}
  trigger={<Collapsible.Trigger />}
  triggerClassName=""
  triggerDisabled={false}
  triggerOpenedClassName=""
  triggerSibling={null}
  triggerStyle={null}
  triggerTagName="span"
>
  <Collapsible.Body />
</Collapsible>
`;
